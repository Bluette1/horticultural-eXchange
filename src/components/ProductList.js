import { useSelector } from "react-redux";
import Plant from "./Plant";
import uuid from "react-uuid";
import Filter from "./Filter";

const ProductList = () => {
  const { filter: category } = useSelector((state) => state.filter);
  const products = useSelector((state) => state.product);
  const categoryPrdcts = (category) =>
    products.filter(
      (prdct) => prdct.category.toLowerCase() === category.toLowerCase()
    );
  const ctprdcts = category === 'All plants' ? products: categoryPrdcts(category);

  return (
    <div className="container d-flex">
      <Filter />
      <>
        {ctprdcts && ctprdcts.length > 0 ? (
          <div className="row d-flex">
            {" "}
            {ctprdcts.map((product) => (
              <Plant key={`plant-${uuid()}`} plant={product} />
            ))}
          </div>
        ) : (
          <></>
        )}
      </>
    </div>
  );
};

export default ProductList;
