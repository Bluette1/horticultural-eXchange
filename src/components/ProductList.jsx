import { useSelector } from 'react-redux';
import uuid from 'react-uuid';
import Plant from './ProductItem';
import Filter from './CategoryFilter';

const ProductList = () => {
  const { filter: category } = useSelector((state) => state.filter);
  const products = useSelector((state) => state.product);
  const categoryPrdcts = (category) => products.filter(
    (prdct) => prdct.category.toLowerCase() === category.toLowerCase(),
  );
  const ctprdcts = category === 'All Plants' ? products : categoryPrdcts(category);

  return (
    <div className="container d-flex">
      <Filter />
      <>
        {ctprdcts && ctprdcts.length > 0 ? (
          <div className="row d-flex">
            {' '}
            {ctprdcts.map((product) => (
              <Plant key={`plant-${uuid()}`} plant={product} />
            ))}
          </div>
        ) : (
          <></>
        )}
      </>
    </div>
  );
};

export default ProductList;
