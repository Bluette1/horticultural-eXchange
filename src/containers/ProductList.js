import { useSelector } from 'react-redux';
import uuid from 'react-uuid';
import Plant from '../components/ProductItem';
import Filter from '../components/CategoryFilter';

const ProductList = () => {
  const { filter: category } = useSelector((state) => state.filter);
  const products = useSelector((state) => state.product);
  const categoryPrdcts = (category) => products.filter(
    (prdct) => prdct.category.toLowerCase() === category.toLowerCase(),
  );
  const ctprdcts = category === 'All Plants' ? products : categoryPrdcts(category);

  return (
    <div className="container d-lg-flex">
      <Filter />
      <>
        {ctprdcts && ctprdcts.length > 0 ? (
          <div className="row d-flex pt-lg-0 pt-3">
            {' '}
            {ctprdcts.map((product) => (
              <Plant key={`plant-${uuid()}`} plant={product} />
            ))}
          </div>
        ) : (
          <></>
        )}
      </>
    </div>
  );
};

export default ProductList;
