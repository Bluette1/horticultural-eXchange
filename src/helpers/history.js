import { createBrowserHistory } from 'history'; // eslint-disable import/no-extraneous-dependencies

const history = createBrowserHistory();

export default history;
