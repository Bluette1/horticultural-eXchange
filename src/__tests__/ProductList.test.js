import React from 'react';
import {
  render, within, waitFor, screen,
} from '@testing-library/react';
import { Router } from 'react-router-dom';
import '@testing-library/jest-dom/extend-expect';
import { Provider } from 'react-redux';
import history from '../helpers/history';
import ProductList from '../containers/ProductList';
import configureTestStore from '../testutils/ConfigureStore';

const plants = [
  {
    id: 64,
    name: 'Aloe Plant',
    category: 'Clearance Sale',
    price: 70,
    care: '',
    created_at: '2021-08-25 17:06:06.393236000 +0000',
    updated_at: '2021-08-25 17:06:06.393236000 +0000',
    image_url: 'https://plant-xchange-app-dev.s3.amazonaws.com/dev...',
    in_stock: true,
    common_name: '',
    description: '',
  },
  {
    id: 65,
    name: 'Protea',
    category: 'Fynbos',
    price: 72,
    care: '',
    created_at: '2021-08-25 17:06:06.393236000 +0000',
    updated_at: '2021-08-25 17:06:06.393236000 +0000',
    image_url: 'https://plant-xchange-app-dev.s3.amazonaws.com/dev...',
    in_stock: true,
    common_name: '',
    description: '',
  },
  {
    id: 35,
    name: 'Erica Tresco',
    category: 'Full Sun Plants',
    price: 65,
    care: '',
    created_at: '2021-08-15 02:08:51.169442000 +0000',
    updated_at: '2021-08-18 08:29:06.965483000 +0000',
    image_url: 'https://plant-xchange-app-dev.s3.amazonaws.com/dev...',
    in_stock: false,
    common_name: '',
    description: '',
  },
  {
    id: 44,
    name: 'Another Plant',
    category: 'Outdoor Garden Plants',
    price: 70,
    care: '',
    created_at: '2021-08-19 14:36:49.572585000 +0000',
    updated_at: '2021-08-19 14:36:49.572585000 +0000',
    image_url: 'https://plant-xchange-app-dev.s3.amazonaws.com/dev...',
    in_stock: false,
    common_name: '',
    description: '',
  },
];

const state = {
  filter: {
    filter: 'Clearance Sale',
  },
  product: plants,
};

const store = configureTestStore(state);
const ProductListWithStore = () => (
  <Provider store={store}>
    <React.StrictMode>
      <Router history={history}>
        <ProductList />
      </Router>
    </React.StrictMode>
  </Provider>
);
test('renders the app - navbar links are displayed correctly', async () => {
  const { container } = render(<ProductListWithStore />);
  await waitFor(() => {
    const productlistContainer = screen.getByTestId('productlist-container');
    expect(productlistContainer).toBeTruthy();
    const categoryFilter = screen.getByTestId('category-filter');
    expect(categoryFilter).toBeTruthy();
    expect(within(container).getByText(/Aloe Plant/i));
    expect(within(container).getByText(/70/i));
    expect(container).toMatchSnapshot();
  });
});
