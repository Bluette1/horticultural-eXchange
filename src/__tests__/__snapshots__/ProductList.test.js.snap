// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the app - navbar links are displayed correctly 1`] = `
<div>
  <div
    class="container d-lg-flex"
    data-testid="productlist-container"
  >
    <div
      class="col-md-3"
      data-testid="category-filter"
    >
      <label
        for="category-select"
      >
        PRODUCT CATEGORIES:
        <select
          id="category-select"
          name="categories"
        >
          <option
            value=""
          >
             Select a category
          </option>
          <option
            value="All Plants"
          >
            All Plants
          </option>
        </select>
      </label>
    </div>
    <div
      class="row d-flex pt-lg-0 pt-3"
    >
       
      <div
        class="col-sm-6 col-md-4"
        role="presentation"
      >
        <div
          class="plant d-flex justify-content center flex-column"
        >
          <button
            class="view btn-primary mb-2"
            data-testid="action-button"
            type="button"
          >
            View item
          </button>
          <img
            alt="plant"
            src="https://plant-xchange-app-dev.s3.amazonaws.com/dev..."
          />
        </div>
        <h4>
          ALOE PLANT
        </h4>
        <h4>
          R
          70
        </h4>
      </div>
    </div>
  </div>
</div>
`;
